src/main/java/edu/ucsd/workflow/RemoveRedundantTSVRows.java

```diff
@@ -0,0 +1,220 @@
+package edu.ucsd.workflow;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+
+public class RemoveRedundantTSVRows
+{
+	/*========================================================================
+	 * Constants
+	 *========================================================================*/
+	private static final String USAGE = "java -jar CCMSWorkflowUtils.jar" +
+		"\n\t-input <InputFile> (TSV file with header)" +
+		"\n\t-output <OutputFile> (Input file with redundant rows removed)" +
+		"\n\t-columns <ColumnHeaders> (Comma-separated list of columns " +
+		"for which only the first row per unique combination of these " +
+		"columns should be retained)";
+
+	/*========================================================================
+	 * Public interface methods
+	 *========================================================================*/
+	public static void main(String[] args) {
+		RemoveRedundantTSVRowsOperation removeRedundantTSVRows = extractArguments(args);
+		if (removeRedundantTSVRows == null)
+			die(USAGE);
+		// remove redundant rows in the TSV
+		BufferedReader input = null;
+		PrintWriter output = null;
+		Throwable thrownError = null;
+		int linesInInputFile = 1;
+		int linesInOutputFile = 1;
+		try {
+			input = new BufferedReader(new FileReader(
+				removeRedundantTSVRows.inputFile));
+			// set up output file writer
+			output = new PrintWriter(removeRedundantTSVRows.outputFile);
+			System.out.println(String.format(
+				"Writing the non-redundant rows in file \"%s\" to output file \"%s\".",
+				removeRedundantTSVRows.inputFile,
+				removeRedundantTSVRows.outputFile));
+			// read header
+			String headerStr = input.readLine();
+			String[] header = headerStr.trim().split("\t");
+			String columnsStr = "";
+			Set<List<String>> uniqueLineIDs = new HashSet<>();
```

Comment:
Your HashSet declaration is not properly typed. Type inference is supported in JRE 1.7 and later, but most of our code is still compliant only with 1.6. I guess this isn't really a problem, since we really ought to come out of our cave and start updating things for Java versions that have been out for almost a decade. Just something I noticed.